zebra-state/src/lib.rs

```diff
@@ -46,36 +45,47 @@ const MAX_BLOCK_REORG_HEIGHT: block::Height = block::Height(MIN_TRASPARENT_COINB
 #[derive(Clone, Debug, Deserialize, Serialize)]
 #[serde(deny_unknown_fields, default)]
 pub struct Config {
-    /// The root directory for storing cached data.
-    ///
-    /// Cached data includes any state that can be replicated from the network
-    /// (e.g., the chain state, the blocks, the UTXO set, etc.). It does *not*
-    /// include private data that cannot be replicated from the network, such as
-    /// wallet data.  That data is not handled by `zebra-state`.
-    ///
-    /// Each network has a separate state, which is stored in "mainnet/state"
-    /// and "testnet/state" subdirectories.
-    ///
-    /// The default directory is platform dependent, based on
-    /// [`dirs::cache_dir()`](https://docs.rs/dirs/3.0.1/dirs/fn.cache_dir.html):
-    ///
-    /// |Platform | Value                                           | Example                            |
-    /// | ------- | ----------------------------------------------- | ---------------------------------- |
-    /// | Linux   | `$XDG_CACHE_HOME/zebra` or `$HOME/.cache/zebra` | /home/alice/.cache/zebra           |
-    /// | macOS   | `$HOME/Library/Caches/zebra`                    | /Users/Alice/Library/Caches/zebra  |
-    /// | Windows | `{FOLDERID_LocalAppData}\zebra`                 | C:\Users\Alice\AppData\Local\zebra |
-    /// | Other   | `std::env::current_dir()/cache`                 |                                    |
-    pub cache_dir: PathBuf,
+    /// The setting for how data should be stored
+    pub storage: StorageMode,
```

Comment:
When switching a bunch of tests from using `in_memory` to `on_disk` I ended up needing to make a bunch of ephemeral configs and I didn't really like that we could have a cache_dir set while its in ephemeral mode so I switched things around to try to make it impossible to represent a misconfiguration but now it won't serialize with toml so I need to figure that out :roll_eyes:

Reply:
@yaahc perhaps you're running into the same toml serialization bug as zebrad did, there's a comment in the `generate` source code that might be relevant.

Reply:
That's what I thought at first too but the error was coming from generate.rs when it tries to convert the config to a toml object and so I think it might just be an incompatibility with the toml spec where it cannot handle enums

Reply:
Ah, hmm, if it's not that one then I'm all out of ideas. Your idea seems plausible to me.

Reply:
I think it'd be fine to just ignore the `cache_dir` value if `ephemeral` is `true`; I agree that it would be nicer to have an enum-style representation but I'm not sure it's worth the trouble of figuring out why this has gone wrong.

Reply:
agree, I'm gonna look into it a little to see if it's possible to keep the enum but if not I'll revert back and just switch the cache_dir to be optional to better reflect how it's used

Reply:
One upside to doing things the simpler way is that test code only has to set `ephemeral = true` and there's no need for a `Config::ephemeral()` method.